### ISSUE TITLE
Carte is not configurable

### ISSUE THREAD

**Issue by HiromuHota** [HiromuHota/pentaho-kettle#90]
Carte runs next to webSpoon (e.g., `http://localhost:8080/spoon/kettle/status`), but it is not configurable.
One drawback is that Data Services cannot be used because a repository cannot be configured (always empty for the Repository name as shown in picture below).

![screen shot 2018-01-12 at 11 01 36 am](https://user-images.githubusercontent.com/20668349/34890815-418d11e8-f788-11e7-8c73-8c21515cc3ab.png)

**Comment by HiromuHota** [comment]
Looks like PentahoServer configures Carte in `org.pentaho.platform.plugin.action.kettle.KettleSystemListener`

```
      String slaveServerConfigFilename =
          "system" + File.separator + "kettle" + File.separator + "slave-server-config.xml"; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
      File slaveServerConfigFile =
          new File( PentahoSystem.getApplicationContext().getSolutionPath( slaveServerConfigFilename ) );
      if ( slaveServerConfigFile.exists() ) {
        InputStream is = new FileInputStream( slaveServerConfigFile );
        Node configNode = getSlaveServerConfigNode( is );
        SlaveServerConfig config = new DIServerConfig( new LogChannel( "Slave server config" ), configNode );
        config.setFilename( slaveServerConfigFile.getAbsolutePath() );
        SlaveServer slaveServer = new SlaveServer();
        config.setSlaveServer( slaveServer );
        CarteSingleton.setSlaveServerConfig( config );
      }
```

This is described in https://help.pentaho.com/Documentation/6.0/0P0/0U0/0A0/010/000.

**Comment by bmolleman** [comment]
Thank you for adding this issue.

**Comment by HiromuHota** [comment]
Resolved by 4dcb443a5986a1dc24df9dd982102ec8b79cd0f2.
@bmolleman please check it whenever you have time.

**Comment by bmolleman** [comment]
It works! Thx. I added the slave-server-config.xml, created a virtual table in WebSpoon and then I queried the virtual table.
